src/typescript/Scala.tmLanguage.ts

```diff
@@ -1165,6 +1168,18 @@ export const scalaTmLanguage: TmLanguage = {
         }
       ]
     },
+    derives: {
```

Comment:
Derives is a soft keyword, wouldn't it be an issue when someone uses it as an identifier?

Reply:
Yeah, it would be a problem.

> derives, if it appears after an extension clause or after the name and possibly parameters of a class, trait, object, or enum definition.
> https://docs.scala-lang.org/scala3/reference/soft-modifier.html

However, I don't think there's a way for regex to check the previous clause is "extension clause or after the name and possibly parameters of a class, trait, object, or enum definition."
(for `using`, we could determine it's after `(` by positive-lookahead though).